### PR TITLE
fix: Align lines correctly when passing `faded_line_ratio` property to NumberPlane (#2894)

### LINKED ISSUE #2894: Unnatural axis alignment when using faded_line_ratio property
## Description of bug / unexpected behaviour

An uneven interval between the axis and grids causes mapping of coordinates using the NumberPlane.c2p unexpected.

## Expected behaviour

The interval between the units from ORIGIN should be uniform and Coordinates should be mapped properly.

## How to reproduce the issue

<details><summary>Code for reproducing the problem</summary>

```py
class Test(Scene):
    def construct(self):
        numberplane = NumberPlane(
            faded_line_ratio=5,
            axis_config={
            "font_size": 40,
            }
        ).add_coordinates()
        coord = numberplane.c2p

        self.add(
            Dot(coord(0, 0)),
            Dot(coord(1, 0)),
            Dot(coord(2, 0)),
        )
        self.add(numberplane)
```

</details>

## Additional media files

<details><summary>Images/GIFs</summary>

image
</details>

## System specifications
Not relevant

### PR DESCRIPTION
**Pull request by imaphatduc**
## Overview: What does this pull request change?
Correctly align lines when creating a new NumberPlane object with `faded_line_ratio` property.

### Before
image

### After
image

## Further Information and Comments
I use the code that #2894 provided.

Closes #2894.

## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

**Review by mohammedbelgoumri — APPROVED** [review]

**Comment by hemanta212** [comment]
Thank you so much for this fix @imaphatduc

File: manim/mobject/graphing/coordinate_systems.py [hunk 1]
```diff
@@ -2661,7 +2661,10 @@ def _get_lines_parallel_to_axis(
             for k, x in enumerate(inputs):
                 new_line = line.copy()
                 new_line.shift(unit_vector_axis_perp_to * x)
-                if k % ratio_faded_lines == 0:
+                if len(inputs) == 0:
+                    lines1.add(new_line)
+                    break
+                elif (k + 1) % ratio_faded_lines == 0:
```
Comment (behackl) [inline manim/mobject/graphing/coordinate_systems.py:2667]: This block cannot be entered: if `len(inputs) == 0`, then `enumerate(inputs)` is the empty list, and as a consequence the for-loop is just empty. Or am I missing something?

```suggestion
                if (k + 1) % ratio_faded_lines == 0:
```

**Review by behackl — APPROVED** [review]

**Comment by imaphatduc** [comment]
@behackl Thank you very much for reviewing my PR. Wow, it's really rewarding to be a contributor of a great math visualization library 🎉. Just one question, why the tests failed after I first committed my changes? (the first commit on this PR)

**Comment by behackl** [comment]
> Which commands are needed to run to generate the control data to a binary file?

See our developer guidelines regarding graphical unit tests: https://docs.manim.community/en/stable/contributing/testing.html#graphical-unit-test -- TL;DR: `pytest --set_test path/to/test_file.py::test_function_name`

> And this binary file is compiled from which source file(s)?

The test functions in the files in `tests/graphical_unit_tests/test_*.py`.

> If in the future I want to contribute to ManimCE, what do I need to do before opening a PR? There seems to be no documentation referring these steps.

You did well with this PR, no need to change anything about your approach, really. There are some comprehensive instructions here: https://docs.manim.community/en/stable/contributing/development.html -- but if you have specific questions it is often easier to just ask in the [#dev-chat channel of our Discord](https://manim.community/discord).

**Comment by imaphatduc** [comment]
Great insight! Thank you.

**Pull request merged**